src/sonic-config-engine/tests/test_multinpu_cfggen.py

```diff
@@ -80,68 +81,73 @@ def test_dummy_run(self):
+        argument = "-v \"DEVICE_METADATA[\'localhost\'][\'hwsku\']\" -m \"{}\"".format(self.sample_graph)
         for asic in range(NUM_ASIC):
-            output = self.run_script_for_asic(argument, asic)
+            output = self.run_script_for_asic(argument, asic, self.port_config[asic])
             self.assertEqual(output.strip(), SKU)
 
     def test_print_data(self):
-        argument = "-m \"{}\" --print-data".format(self.sample_graph)
+        argument = "-m \"{}\" -p \"{}\" --print-data".format(self.sample_graph, self.sample_port_config)
         output = self.run_script(argument)
         self.assertGreater(len(output.strip()) , 0)
+        argument = "-m \"{}\" --print-data".format(self.sample_graph)
         for asic in range(NUM_ASIC):
-            output = self.run_script_for_asic(argument, asic)
+            output = self.run_script_for_asic(argument, asic, self.port_config[asic])
             self.assertGreater(len(output.strip()) , 0)
 
     def test_additional_json_data(self):
         argument = '-a \'{"key1":"value1"}\' -v key1'
         output = self.run_script(argument)
         self.assertEqual(output.strip(), 'value1')
         for asic in range(NUM_ASIC):
-            output = self.run_script_for_asic(argument, asic)
+            output = self.run_script_for_asic(argument, asic, self.port_config[asic])
             self.assertEqual(output.strip(), 'value1')
 
     def test_read_yaml(self):
         argument = '-v yml_item -y ' + os.path.join(self.test_dir, 'test.yml')
         output = yaml.load(self.run_script(argument))
         self.assertListEqual(output, ['value1', 'value2'])
         for asic in range(NUM_ASIC):
-            output = yaml.load(self.run_script_for_asic(argument, asic))
+            output = yaml.load(self.run_script_for_asic(argument, asic, self.port_config[asic]))
             self.assertListEqual(output, ['value1', 'value2'])
 
     def test_render_template(self):
         argument = '-y ' + os.path.join(self.test_dir, 'test.yml') + ' -t ' + os.path.join(self.test_dir, 'test.j2')
         output = self.run_script(argument)
         self.assertEqual(output.strip(), 'value1\nvalue2')
         for asic in range(NUM_ASIC):
-            output = self.run_script_for_asic(argument, asic)
+            output = self.run_script_for_asic(argument, asic, self.port_config[asic])
             self.assertEqual(output.strip(), 'value1\nvalue2')
 
     def test_metadata_tacacs(self):
-        argument = '-m "' + self.sample_graph +  '" --var-json "TACPLUS_SERVER"'
+        argument = '-m "' + self.sample_graph + '" -p "' + self.sample_port_config +  '" --var-json "TACPLUS_SERVER"'
         output = json.loads(self.run_script(argument))
         self.assertDictEqual(output, {'123.46.98.21': {'priority': '1', 'tcp_port': '49'}})
         #TACPLUS_SERVER not present in the asic configuration.
+        argument = '-m "' + self.sample_graph +  '" --var-json "TACPLUS_SERVER"'
```

Comment:
>argument [](http://example.com/codeflow?start=8&length=8)

If you do not add this line, will it fail?

Reply:
I add port config in line 126, and below line will also add port config:
output = json.loads(self.run_script_for_asic(argument, asic, self.port_config[asic]))
If don't add line 130, there will be two port config for the same command.